### PR TITLE
Alerting: Store sensitive settings encrypted for Telegram notifier

### PR DESCRIPTION
**Pull request by wbrowne**
**What this PR does / why we need it**:
We need to update all alert notifiers so that they support storing sensitive information encrypted. This PR covers Telegram 

**Which issue(s) this PR fixes**:
(part of) https://github.com/grafana/grafana/issues/25967

### CODE REVIEW

**Review by aknuds1 — APPROVED** [review]
Looks good AFAICT.

File: pkg/services/alerting/notifiers/telegram.go [hunk 1]
```diff
@@ -86,7 +96,7 @@ func NewTelegramNotifier(model *models.AlertNotification) (alerting.Notifier, er
 		return nil, alerting.ValidationError{Reason: "No Settings Supplied"}
 	}
 
-	botToken := model.Settings.Get("bottoken").MustString()
+	botToken := model.DecryptedValue("bottoken", model.Settings.Get("bottoken").MustString())
 	chatID := model.Settings.Get("chatid").MustString()
 	uploadImage := model.Settings.Get("uploadImage").MustBool()
```
[the remark this answers is not in the document]
Comment (marefr) [reply pkg/services/alerting/notifiers/telegram.go:101]: `uploadImage` is a setting available to all notifiers, but not necessarily used/supported by all of them . Seems like this notifier still are reading setting where probably all other notifiers are using the base implementation method `NeedsImage()` instead. See example usage here: https://github.com/grafana/grafana/blob/954a2811b3f00bb5d4d0b40426bf3daedaf50215/pkg/services/alerting/notifiers/slack.go#L391

Suggest adding back the `uploadImage` to provisioning documentation.

File: docs/sources/administration/provisioning.md [hunk 2]
```diff
@@ -520,11 +520,10 @@ The following sections detail the supported settings and secure settings for eac
 
 #### Alert notification `telegram`
 
-| Name        |
-| ----------- |
-| bottoken    |
-| chatid      |
-| uploadImage |
```
[the remark this answers is not in the document]
Comment (marefr) [reply docs/sources/administration/provisioning.md:527]: See my other comment. `uploadImage` This change needs to be reverted.

**Review by marefr — CHANGES_REQUESTED** [review]
See comments

File: pkg/services/alerting/notifiers/telegram.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wbrowne) [reply pkg/services/alerting/notifiers/telegram.go:101]: Makes sense. Have added back the field to the docs.

This sounds like something we can tackle separately if necessary as it's also missing from the Pushover notifier

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Just need to sync/merge with master to resolve CI problems (I think)

**Pull request merged**